AWSAppSyncClient/Internal/SubscriptionWithSyncConfiguration.swift

```diff
@@ -0,0 +1,29 @@
+//
+// Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+/// Configuration values for an AppSyncSubscriptionWithSync instance.
+public struct SubscriptionWithSyncConfiguration {
```

Comment:
I think this not clear naming. I thought that it's new `Subscription` class at once, but it's configuration of `SubscriptionWithSync`.

Reply:
I think that the old name is much clearer.

Reply:
Thanks, re-looking at this with a day of distance, I agree that the [new name is a bit crap](https://martinfowler.com/bliki/TwoHardThings.html). :) However, as much as I like the existing name, I don't want to change it to a `struct` since that would break anybody relying on the existing class semantics of that type. (It's unlikely, but I don't want to break an API contract without warning.) Let me consider this.

Reply:
OK, @rohandubal convinced me--I'll convert `SyncConfiguration` to a `struct` rather than deprecating the previous type. That will make this PR a breaking change, but we expect the actual impact on developers to be very small since this is a relatively new API, and it would be a pretty uncommon pattern to be relying on class semantics for this particular type.

Reply:
I also think that it's not a big deal replacing `struct` with `class`.

1. I'm not sure that there're a lot users who use AWSAppSync iOS SDK and the latest feature `sync`.
2. There weren't Swift versions that don't have breaking changes. So I think this isn't new for Swift developers :)